lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -760,7 +760,10 @@ def _on_set_password_action(self, event: ActionEvent):
         if self.upgrade_in_progress:
             event.fail("Setting password not supported while upgrade in-progress")
             return
-        if self.opensearch_peer_cm.deployment_desc().typ != DeploymentType.MAIN_ORCHESTRATOR:
+        if (
+            not self.opensearch_peer_cm.deployment_desc()
+            or self.opensearch_peer_cm.deployment_desc().typ != DeploymentType.MAIN_ORCHESTRATOR
+        ):
             event.fail("The action can be run only on the leader unit of the main cluster.")
```

Comment:
nit: make phrasing consistent with the other messages
```suggestion
            event.fail("The action can only be run on the leader unit of the main cluster.")
```

Reply:
Ah! Good catch! I copied the same message